subsys/net/conn_mgr/online_checker.c

```diff
@@ -801,7 +825,27 @@ void conn_mgr_online_connectivity_check(void)
 			 * generate CONNECTIVITY_ONLINE event. The online
 			 * check might take some time.
 			 */
+			k_mutex_lock(&check_running_lock, K_FOREVER);
+			if (check_running) {
+				NET_DBG("Online check already running");
+				k_mutex_unlock(&check_running_lock);
+				break;
+			}
+
+			check_running = true;
+			k_mutex_unlock(&check_running_lock);
```

Comment:
Perhaps we should unlock `check_running_lock` after setting `states[i] |= CONN_MGR_IF_ONLINE_CHECK;`

Reply:
I will leave the `check_running = true` there so that it is protected by the lock. Probably does not make much difference as you said.